File: 0525-contiguous-array/0525-contiguous-array.java
class Solution {
    public int findMaxLength(int[] nums) {
            Map<Integer,Integer> map=new HashMap<>();
            map.put(0,-1);
            int maxLen=0;
            int counter=0;
            int n=nums.length;
            for(int index=0;index<n;index++){
                if(nums[index]==1){
                    counter++;
                }
                else{
                    counter--;
                }
                if(map.containsKey(counter)){
                    int currLen=index-map.get(counter);
                    maxLen=Math.max(maxLen,currLen);
                }
                else map.put(counter,index);
                }
            return maxLen;
    }
}

File: 0525-contiguous-array/0525-contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        maxlen=0
        values={0:-1}
        counter=0
        for i,num in enumerate(nums):
            counter+=1 if num==0 else -1
            if counter in values:
                maxlen=max(maxlen,i-values[counter])
            else:
                values[counter]=i
        return maxlen